File: src/main/java/tfc/project/ultimateComboRoutes/model/services/ComboService.java
package tfc.project.ultimateComboRoutes.model.services;

import java.util.ArrayList;

import tfc.project.ultimateComboRoutes.model.entities.Combo;
import tfc.project.ultimateComboRoutes.model.exceptions.InstanceNotFoundException;

public interface ComboService {

	Combo addCombo(Long adminId, Long characterId, String secuence, String difficulty, int totalDamage, String demo)
			throws InstanceNotFoundException;

	Combo editCombo(Long comboId, String secuence, String difficulty, int totalDamage, String demo)
			throws InstanceNotFoundException;

	ArrayList<Combo> showCharacterCombos(Long characterId) throws InstanceNotFoundException;

	ArrayList<Combo> showCharacterCombosByDifficulty(Long characterId, String difficulty)
			throws InstanceNotFoundException;
}

File: frontend/src/backend/comboService.js
import {
    fetchConfig,
    appFetch,
}from "./appFetch";

export const addCombo = (
    characterId,
    secuence,
    difficulty,
    totalDamage,
    demo,
    onSuccess,
    onErrors
) => 
    appFetch(
        `/combos/${characterId}/addCombo`, fetchConfig("POST", {characterId, secuence, 
            difficulty, totalDamage, demo}),  
            onSuccess,
            onErrors
        );

export const editCombo = (
    id,
    secuence,
    difficulty,
    totalDamage,
    demo,
    onSuccess,
    onErrors
) =>
    appFetch(
        `/combos/${id}/editCombo`, fetchConfig("PUT", {secuence, difficulty,
            totalDamage, demo}),
            onSuccess,
            onErrors
        );

export const showCharacterCombos = (
    id,
    onSuccess
) =>
    appFetch(
        `/combos/${id}`, fetchConfig("GET"),
        onSuccess
    );

export const filterByDifficulty = (
    {difficulty, characterId},
    onSuccess
) => 
    appFetch(
        `/combos/${characterId}/filterByDifficulty?difficulty=${difficulty}`, fetchConfig("GET"),
        onSuccess
    );
    